third_party/e2e/engine.cc

```diff
@@ -0,0 +1,90 @@
+#include "engine.h"
```

Comment:
Is there a specific reason why these files and in general the e2e directory is under third_party? Is it just that this is not part of the library?

Reply:
If you have time to do so, do move out of third party anything that was developed by the team.

Reply:
I think this does actually belong in `third_party` as it's logically part of the kernel modules, which themselves must be in third_party due to different licensing requirements than the rest of the project.